tests/appsec/iast/test_iast_no_request.py

```diff
@@ -0,0 +1,23 @@
+# Unless explicitly stated otherwise all files in this repository are licensed under the the Apache License Version 2.0.
+# This product includes software developed at Datadog (https://www.datadoghq.com/).
+# Copyright 2021 Datadog, Inc.
+
+from utils import missing_feature, interfaces, coverage, released
+
+
+@released(dotnet="?", golang="?", java="?", nodejs="?")
+@released(php_appsec="?", python="?", ruby="?", cpp="?")
+@coverage.basic
+class TestIastNoRequest:
+    """ IAST deals with vulnerabilities outside of requests """
+
+    @missing_feature(reason="Need to implement span creation")
+    def test_span_creation(self):
```

Comment:
This test will get all root spans with `_dd.iast.json`. If at least one of then contains `type==vulnerability`, it'll be ok. Otherwise it fails.

It also means that if there is no `iast` event reported, it fails.

If it's what you want to achieve, it may be a good idea to add a method that trigger a iast event no ?

Reply:
Yes, that's exactly what the tests does. Regarding the trigger of the iast event it's done during the startup of the app (check `CriptyExamples.java`)

```java
    @PostConstruct
    public void setup() throws NoSuchAlgorithmException {
        Security.addProvider(new BouncyCastleProvider());
        final MessageDigest md4 = MessageDigest.getInstance("md4");
        consumeMessageDigest(md4, "vulnerability during startup");
    }
```

We could try to trigger it during a request but outside of an APM span, but it will only make the code a bit harder, do you think we should we try it?

Reply:
Actually, there are some existing endpoints that trigger IAST event, look for `/iast/` endpoints in the code.

So you just have to add two code line in python test code, rather adding some code at startup for **all** weblogs (more than 30 now).

Reply:
I've updated the description of the PR as it was probably not very clear, what this PR tries to solve cannot be accomplished via python code alone as all vulnerabilities triggered from `/iast/` are linked to requests and they don't produce custom spans (the information is always attached to the current trace segment).  We are testing the other cases like startup code or background jobs, what I mentioned before is that instead of a vulnerability triggered by startup code we could start a background job from a request and then wait for the vulnerability to appear (It might be tricky for some languages like nodejs but it can be done).